vnext/Scripts/Microsoft.ReactNative.VersionCheck.targets

```diff
@@ -0,0 +1,100 @@
+          // System.Text.Json or NewtonSoft would require a package depednency, which is pretty cumbersome today.
+          var jsonSerializer = new DataContractJsonSerializer(typeof(PackageJson));
+          using (var fileStream = new FileStream(PackageJsonPath, FileMode.Open, FileAccess.Read, FileShare.Read))
+          {
+            var pkgJson = (PackageJson)jsonSerializer.ReadObject(fileStream);
+            NpmName = pkgJson.name;
+            NpmVersion = pkgJson.version;
+          }
+
+          return true;
+        }
+    }
+]]>
+      </Code>
+    </Task>
+  </UsingTask>
+  
+  <!-- This target checks if the npm package version matches the nuget package -->
+  <Target 
+    Name="ReactNativeWindowsVersionCheck" 
+    BeforeTargets="PrepareForBuild"
+    Condition="'$(ReactNativeWindowsAllowNugetNpmMismatch)' != 'true'"
+    >
+
+    <PropertyGroup>
+      <_ReactNativeWindowsVersionCheckNugetVersion>$$nuGetPackageVersion$$</_ReactNativeWindowsVersionCheckNugetVersion>
+    </PropertyGroup>
+   
+    <!-- Validate package.json file -->
+    <Error
+      Condition="'$(ReactNativeWindowsDir)' == ''"
+      Text="The project does not defined property 'ReactNativeWindowsDir'. This is required to be configured for any react-native-windows project."
+    />
+    
+    <PropertyGroup>
+      <_ReactNativeWindowsVersionCheckPackageJsonPath>$([System.IO.Path]::Combine($(ReactNativeWindowsDir), 'package.json'))</_ReactNativeWindowsVersionCheckPackageJsonPath>
+    </PropertyGroup>
+    
+    <Error
+      Condition="!Exists($(_ReactNativeWindowsVersionCheckPackageJsonPath))"
+      Text="The package.json file for react-native-windows could not be found at: '$(_ReactNativeWindowsVersionCheckPackageJsonPath)'"
+    />
+
+    <ExtractNpmVersion PackageJsonPath="$(_ReactNativeWindowsVersionCheckPackageJsonPath)">
+      <Output PropertyName="_ReactNativeWindowsVersionCheckNpmVersion" TaskParameter="NpmVersion" />
+      <Output PropertyName="_ReactNativeWindowsVersionCheckNpmName" TaskParameter="NpmName" />
+    </ExtractNpmVersion>
+  
+    <Message Text="Discovered package '$(_ReactNativeWindowsVersionCheckNpmName)' with version '$(_ReactNativeWindowsVersionCheckNpmVersion)'" Importance="High" />
+
+    <Error 
+      Condition="'$(_ReactNativeWindowsVersionCheckNpmName)' != 'react-native-windows'"
+      Text="Expected to find npm package 'react-native-windows' in '$(_ReactNativeWindowsVersionCheckPackageJsonPath)', encountered '$(_ReactNativeWindowsVersionCheckNpmName)'" />
+    <Error 
+      Condition="'$(_ReactNativeWindowsVersionCheckNpmVersion)' != '$(_ReactNativeWindowsVersionCheckNugetVersion)'"
+      Text="Mismatch detected between npm package versions and nuget package version.%0a    Npm:   '$(_ReactNativeWindowsVersionCheckNpmVersion)' - $(_ReactNativeWindowsVersionCheckPackageJsonPath)%0a    NuGet: '$(_ReactNativeWindowsVersionCheckNugetVersion)' - $(MSBuildProjectFile)%0aUpdate either to match the version." />
```

Comment:
Consider adding specific instructions to update packages.config, or template project PackageReference (or move that one internal to the package). A lot of the folks using RNW, won't be familiar with the native toolchain, so they'll do an npm upgrade automatically and not know what to do to update the NuGet package.

FWIW I think it's going to be much more likely that the NuGet package needs to be updated than the npm one.

Reply:
I'll create a page on the doc and refer to that one.

Reply:
Update error message is: 
```
      error : Mismatch detected between npm package versions and nuget package version.
      error :     Npm:   '0.0.0-canary.148' - Z:\Temp\tst2\node_modules\react-native-windows\package.json
      error :     NuGet: '0.0.1-MyTest010' - Z:\Temp\tst2\windows\tst2\tsts2.csproj
      error : To update the nuget version, please see https://microsoft.github.io/react-native-windows/docs/nuget-update for instructions
```
The current url for the page is: https://microsoft.github.io/react-native-windows/docs/next/nuget-update